spec/index.md

```diff
@@ -2301,6 +2301,34 @@ This function returns fee percentiles, measured in millisatoshi/vbyte (1000 mill
 
 The [standard nearest-rank estimation method](https://en.wikipedia.org/wiki/Percentile#The_nearest-rank_method), inclusive, with the addition of a 0th percentile is used. Concretely, for any i from 1 to 100, the ith percentile is the fee with rank `⌈i * 100⌉`. The 0th percentile is defined as the smallest fee (excluding coinbase transactions).
 
+### IC method `take_snapshot` {#ic-take_snapshot}
+
+This method takes a snapshot of the canister. Subsequent `take_snapshot` calls will replace the existing snapshot with a new one.
+
+The optional `duration` parameter can be used to limit the storage duration of a snapshot. When provided, the snapshot will be stored for the specified duration (in seconds) and will be deleted automatically at the end of this duration. In this case,
```

Comment:
I think `duration` is a weird name. It sounds like it would be the duration of the `take_snapshot`. Would it maybe be easier to name this `delete_at_timestamp` or `preserve_until_timestamp` or so?

Reply:
Sure, we rename it, `preserve_until_timestamp` sounds better, or `preserve_duration`?

Reply:
I had a discussion today with Ulan. I will remove `duration` altogether. I was informed that there had been some work on `reserved_cycles,` which reserves cycles from the balance for resource allocation for a longer period of time. We will integrate the snapshots with `reserve_cycles` mechanism, so the duration does not make sense anymore. Will update the design.